common/src/chain/config/builder.rs

```diff
@@ -320,6 +324,7 @@ impl Builder {
         Self {
             chain_type,
             bip44_coin_type: chain_type.default_bip44_coin_type(),
+            checkpoints: None,
```

Comment:
I think a better way to represent this is to enforce having genesis in it by default instead of None. So instead of `Option<BTreeMap>`, it becomes `BTreeMap` with the invariant that the first block in it is genesis.

Reply:
> I think a better way to represent this is to enforce having genesis in it by default instead of None. So instead of Option<BTreeMap>, it becomes BTreeMap with the invariant that the first block in it is genesis.

We can't enforce the presence of the genesis here, because we don't yet know what genesis will actually be used (e.g. some tests use a custom genesis).

Reply:
Alrighty.